### ISSUE TITLE
MVT Layer loadOptions not updating

### ISSUE THREAD

**Issue by sprezz13** [visgl/deck.gl#6205]
#### Description
The loadOptions for the MVT Layer are not updating. I have a bearer token that refreshes every 10 minutes, once the token refreshes I re render the layer. However the loadOptions do not pick up the new token and still use the old one.

#### Expected Behavior
The MVT Layer's loadOptions should update and contain the new bearer token after it re renders.

#### Repro Steps
This is an example of the code being used, we have a bearer token we get from our backend and it refreshes every 10 minutes. 
```
new MVTLayer({
          id: 'mvt-layer-accesstoken',
          data: my/api/...,
          loadOptions: {
            fetch: {
                headers: {
                    Authorization: `Bearer ${apiToken}`
                }
            }
        },
})

```

#### Environment
 - Framework Version: Latest deck.gl version (8.5.8)
 - Browser Version: Chrome
 - OS: Max OS v10.15.7

#### Logs
Just 403 error logs because of the expired token.

**Comment by Pessimistress** [comment]
Fixed in v8.5 9.

**Comment by reececasey** [comment]
@Pessimistress before I raise a bug I am guessing this has no relation to the GeoJsonLayer? I am seeing the same issue there? I have not tried v8.5.9 yet I am on v8.5.8 but looking at this PR it looks very much tile related?
